Sources/Client/Client/Client.swift

```diff
@@ -53,38 +53,50 @@ public final class Client {
     var waitingRequests = [WaitingRequest]()
     
     /// A web socket client.
-    lazy var webSocket = WebSocket()
+    lazy var webSocket: WebSocket = {
+        let webSocket = WebSocket()
+        webSocket.onConnect = setupWebSocketOnConnect
+        webSocket.onEvent = setupWebSocketOnEvent
+        return webSocket
+    }()
     
     public var connection: Connection { webSocket.connection }
     
     /// A WebSocket connection callback. This should only be used when you only use the Low-Level Client.
-    public var onConnect: Client.OnConnect = { _ in } {
-        didSet { webSocket.onConnect = setupWebSocketOnConnect }
-    }
+    public var onConnect: Client.OnConnect = { _ in }
     
     /// Saved onConnect for a completion block in `connect()`.
     var savedOnConnect: Client.OnConnect?
     
-    /// A WebSocket events callback. This should only be used when you only use the Low-Level Client.
-    public var onEvent: Client.OnEvent = { _ in } {
-        didSet { webSocket.onEvent = setupWebSocketOnEvent }
+    lazy var eventHandlingQueue: DispatchQueue = {
+        self.webSocket.webSocket.callbackQueue
+    }()
+    
+    var onEventObservers = [String: OnEvent]()
+    lazy var onEvent: Client.OnEvent = { [unowned self] event in
+        self.eventHandlingQueue.async {
+            self.onEventObservers.values.forEach({ $0(event) })
+        }
     }
     
     lazy var urlSession = URLSession(configuration: .default)
     lazy var urlSessionTaskDelegate = ClientURLSessionTaskDelegate() // swiftlint:disable:this weak_delegate
     let callbackQueue: DispatchQueue?
-    private let uuid = UUID()
     
     /// A log manager.
     public let logger: ClientLogger?
     public let logOptions: ClientLogger.Options
     
-    /// An observable user. This should only be used when you only use the Low-Level Client.
-    public var onUserUpdate: OnUpdate<User>?
+    var onUserUpdateObservers = [String: OnUpdate<User>]()
+    lazy var onUserUpdate: OnUpdate<User> = { [unowned self] user in
+        self.eventHandlingQueue.async {
+            self.onUserUpdateObservers.values.forEach({ $0(user) })
+        }
+    }
     
     private(set) lazy var userAtomic = Atomic<User> { [unowned self] newUser, _ in
         if let user = newUser {
```

Comment:
The same here.
```suggestion
    var onUserUpdateObservers = [String: OnUpdate<User>]()
    
    private(set) lazy var userAtomic = Atomic<User> { [unowned self] newUser, _ in
        if let user = newUser {
            self.eventHandlingQueue.async {
                self.onUserUpdateObservers.values.forEach({ $0(user) })
            }
        }
    }
```

Reply:
This is not possible since `onUserUpdate` is called from `User`:
```swift
let unreadCountAtomic = Atomic<UnreadCount>(.noUnread) { _, _ in Client.shared.onUserUpdate(User.current) }
```
an alternative would be:
```swift
let unreadCountAtomic = Atomic<UnreadCount>(.noUnread) { _, _ in Client.shared.userAtomic.set(User.current) }
```
which I fear might have side effects.